### ISSUE TITLE
[java] AccessorMethodGeneration: Name clash with another public field not properly handled

### ISSUE THREAD

**Issue by wkurniawan07** [pmd/pmd#342]
**Rule Set:** AccessorMethodGeneration (Java - Design), PMD 5.5.5

**Description:**

It appears that the rule AccessorMethodGeneration fails to resolve name clash with public field in another class with identical name.

**Code Sample demonstrating the issue:**

```java
public class ClassA {

    public String field;

}
```

```java
public class ClassB {

    private String field;

    private class ClassC {

        public void doSomething() {
            ClassA a = new ClassA();
            a.field = "abcd"; // should not be a violation here but reported as one
        }

    }

}
```

Changing `ClassB#field` to e.g. `ClassB#field0` removes the violation.

**Running PMD through:** Gradle

**Comment by jsotuyod** [comment]
Thanks for the report. 

This is a known limitation. To be able to properly pinpoint which `field` we are referring to requires complete type resolution, which is currently lacking and is not trivial to implement.

Fortunately, [we are planning to work on it in the near future](https://github.com/pmd/pmd/wiki/Project-Ideas-%5BMature%5D#complete-type-resolution-for-java), we even got [proposals for google summer of code to tackle this](https://github.com/pmd/pmd/wiki/Complete-type-resolution-for-Java), so we can expect a proper fix over the course of the year.

I'll keep this open to track the issue and let you know when a fix is ready.

**Comment by wkurniawan07** [comment]
@jsotuyod much appreciated, thanks! 👍👍

**Comment by adangel** [comment]
Fixed via #2899 for PMD 7.
